pip/req/req_install.py

```diff
@@ -231,27 +237,23 @@ def build_location(self, build_dir):
         # FIXME: Is there a better place to create the build_dir? (hg and bzr
         # need this)
         if not os.path.exists(build_dir):
+            logger.debug('Creating directory %s', build_dir)
             _make_build_dir(build_dir)
         return os.path.join(build_dir, name)
 
     def correct_build_location(self):
         """If the build location was a temporary directory, this will move it
-        to a new more permanent location"""
+        to a new more permanent location
+        Move self._temp_build_dir to self._ideal_build_dir/self.name
+        """
         if self.source_dir is not None:
             return
         assert self.req is not None
         assert self._temp_build_dir
+        assert self._ideal_build_dir
```

Comment:
_correct_build_location is only called by pip and those assert are (I guess ?) mainly here to avoid the use of the method in the wrong context.
I think we can assume that at least pip tests won't be run with `-O`.